src/NLog/Config/XmlLoggingConfiguration.cs

```diff
@@ -915,70 +915,36 @@ private void ParseExtensionsElement(NLogXmlElement extensionsElement, string bas
                 string assemblyFile = addElement.GetOptionalAttribute("assemblyFile", null);
                 if (assemblyFile != null)
                 {
-                    try
-                    {
 #if SILVERLIGHT && !WINDOWS_PHONE
-                                var si = Application.GetResourceStream(new Uri(assemblyFile, UriKind.Relative));
-                                var assemblyPart = new AssemblyPart();
-                                Assembly asm = assemblyPart.Load(si.Stream);
+                    var si = Application.GetResourceStream(new Uri(assemblyFile, UriKind.Relative));
+                    var assemblyPart = new AssemblyPart();
+                    Assembly asm = assemblyPart.Load(si.Stream);
 #else
 
-                        string fullFileName = Path.Combine(baseDirectory, assemblyFile);
-                        InternalLogger.Info("Loading assembly file: {0}", fullFileName);
+                    string fullFileName = Path.Combine(baseDirectory, assemblyFile);
+                    InternalLogger.Info("Loading assembly file: {0}", fullFileName);
 
-                        Assembly asm = Assembly.LoadFrom(fullFileName);
+                    Assembly asm = Assembly.LoadFrom(fullFileName);
 #endif
-                        this.ConfigurationItemFactory.RegisterItemsFromAssembly(asm, prefix);
-                    }
-                    catch (Exception exception)
```

Comment:
I think we still need the catch, but with the same logic as the prev change (with my comment)